include/ut_kvp.h

```diff
@@ -0,0 +1,200 @@
+ * 
+ * @param[in] pInstance - Handle to the KVP instance.
+ * @param[in] pszKey - Null-terminated string representing the key to search for.
+ * 
+ * @returns The `uint16_t` value on success, or 0 on error (check logs for details).
+ */
+uint16_t ut_kvp_getUInt16Field(ut_kvp_instance_t *pInstance, const char *pszKey);
+
+/**!
+ * @brief Gets a uint32_t value from the KVP profile.
+ * 
+ * @param[in] pInstance - Handle to the KVP instance.
+ * @param[in] pszKey - Null-terminated string representing the key to search for.
+ * 
+ * @returns The `uint32_t` value on success, or 0 on error (check logs for details).
+ */
+uint32_t ut_kvp_getUInt32Field(ut_kvp_instance_t *pInstance, const char *pszKey);
+
+/**!
+ * @brief Gets a uint64_t value from the KVP profile.
+ * 
+ * @param[in] pInstance - Handle to the KVP instance.
+ * @param[in] pszKey - Null-terminated string representing the key to search for.
+ * 
+ * @returns The `uint64_t` value on success, or 0 on error (check logs for details).
+ */
+uint64_t ut_kvp_getUInt64Field(ut_kvp_instance_t *pInstance, const char *pszKey);
+
+/**!
+ * @brief Retrieves a string value from the KVP profile.
+ * 
+ * The caller owns the memory passed into `pszReturnedString`, which is filled out by this function. 
+ * The `uStringSize` should include space for the null-terminator.
+ *
+ * @param[in] pInstance - Handle to the KVP instance.
+ * @param[in] pszKey - Key of the string value to retrieve (null-terminated string).
+ * @param[out] pszReturnedString - Pre-allocated buffer to store the retrieved string.
+ * @param[in] uStringSize - Size of the `pszReturnedString` buffer (including space for the null-terminator).
+ *
+ * @returns Status of the operation (`ut_kvp_status_t`):
+ * @retval UT_KVP_STATUS_SUCCESS - String value was found and successfully copied to `pszReturnedString`.
+ * @retval UT_KVP_STATUS_INVALID_PARAM - An invalid parameter was passed (e.g. invalid buffer size).
+ * @retval UT_KVP_STATUS_PARSING_ERROR - An error occurred while parsing the KVP data.
+ * @retval UT_KVP_STATUS_NO_DATA - The KVP instance does not contain any data.
+ * @retval UT_KVP_STATUS_KEY_NOT_FOUND - The specified key (`pszKey`) was not found in the KVP data.
+ * @retval UT_KVP_STATUS_NULL_PARAM - A null parameter was passed.
+ * @retval UT_KVP_STATUS_INVALID_INSTANCE - An invalid KVP instance handle was passed.
+ */
+ut_kvp_status_t ut_kvp_getStringField( ut_kvp_instance_t *pInstance, const char *pszKey, char *pszReturnedString, uint32_t uStringSize );
+
+/**
+ * @brief Print the complete kvp data
+ *
+ * Where the data is invalid, no output will occur
+ *
+ * @param pInstance - pointer to the instance to print
+ */
+void ut_kvp_print( ut_kvp_instance_t *pInstance );
+
+/**
+ * @brief Get the data block from the instance
+ *
+ * Return the complete data block as a string
+ *
+ * @param pInstance - pointer to the instance to print
+ */
+uint8_t *ut_kvp_getData( ut_kvp_instance_t *pInstance, uint32_t *pReturnedLength );
```

Comment:
you should remove this function protype and put it in the todo if you've not done it.